config.yaml

```diff
@@ -0,0 +1,25 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+options:
+  port:
+    description: |
+      Prometheus-edge-hub port
+    type: int
+    default: 9091
+  grpc-port:
+    description: |
+      Prometheus-edge-hub grpc-port
+    type: int
+    default: 0
+  limit:
+    description: |
+      Limit the total metrics in the cache at one time. Will reject a push if cache is full.
+      Default is -1 which is no limit. (default -1)
+    type: int
+    default: -1
+  scrape-timeout:
```

Comment:
Wait why is this on the scrapee side? O_o We have the prometheus-scrape-config charm to set these configs

Reply:
Not sure exactly, looking at the code, scrapeTimeout is used in the MetricHub's exposeMetrics method. So when you make a GET to the edge hub's metrics endpoint and it takes too long to build the metrics string, it'll return nothing. And by too long I mean the scrape timeout duration.

Refs: https://github.com/facebookincubator/prometheus-edge-hub/blob/bdd437aab2a9f028087c3d845151c6fce86fc732/hub/hub.go#L241